spec/classes/rabbitmq_spec.rb

```diff
@@ -6,7 +6,7 @@
     let(:facts) {{ :osfamily => 'Unsupported' }}
 
     it 'we fail' do
-      expect { subject }.to raise_error(/not supported on an Unsupported/)
+      expect { is_expected.to compile }.to raise_error(RSpec::Expectations::ExpectationNotMetError, /not supported on an Unsupported/)
```

Comment:
Actually, I think you have the right idea below, by using `catalogue` instead of `subject`. The original line with `subject` replaced by `catalogue` seems to work, and it even seems to work for checking the type of error, if we want. So either

```
expect { catalogue }.to raise_error(/not supported on an Unsupported/)
```

or 

```
expect { catalogue }.to raise_error(Puppet::Error, /not supported on an Unsupported/)
```

are optimal solutions, I think.